### PR TITLE
Add workflow section to results page with metadata for review

### PR DESCRIPTION
**Pull request by duncan-brown**
This pull requests adds a new section to the results page to store data about the workflow.

### CODE REVIEW

**Comment by duncan-brown** [comment]
Currently testing, so not yet ready for merge. Posted for comments from @spxiwh @cmbiwer and @ahnitz.

Example here https://sugar-jobs.phy.syr.edu/~dbrown/orange-grid/sugar-nonsharedfs-test-9/8._workflow/

File: bin/hdfcoinc/pycbc_make_coinc_search_workflow [hunk 1]
```diff
@@ -42,26 +44,42 @@ parser.add_argument("-d", "--output-dir", default=None,
...
-                                 'analysis_time',
+rdir = layout.SectionNumber('results', ['analysis_time',
                                  'detector_sensitivity',
                                  'single_triggers',
                                  'coincident_triggers',
                                  'injections',
                                  'search_sensitivity',
                                  'result',
+                                 'workflow',
                                  ])
```
Comment (ahnitz) [inline bin/hdfcoinc/pycbc_make_coinc_search_workflow]: Is there any reason not to put this under condiguration/workflow? We already have quite a few top-level sections, and I think this fits well under configuration.

File: bin/hdfcoinc/pycbc_make_coinc_search_workflow [hunk 2]
```diff
@@ -387,6 +404,16 @@ if len(files_for_combined_injfind) > 0:
...
+
+# make segment summary and copy segment files into results pages
+analysis_time_summ = ([(time_file,)] + [(seg_summ_plot,)] + 
+                      [(seg_summ_table, veto_summ_table)])
+layout.two_column_layout(rdir['analysis_time'], analysis_time_summ)
+wf.makedir(rdir['analysis_time/segment_data'])
+for seg_file in os.listdir("segments/"):
+    if seg_file.endswith(".xml") or seg_file.endswith(".xml.gz"):
+      shutil.copy(os.path.join("segments",seg_file),
+                  os.path.join(rdir['analysis_time/segment_data'],seg_file))
```
Comment (ahnitz) [inline bin/hdfcoinc/pycbc_make_coinc_search_workflow]: Don't use copy here. Instead use a symlink as we do for the main summary section. That is valid before the files exists, and won't require us to keep the file in two places.

File: bin/hdfcoinc/pycbc_make_coinc_search_workflow (hunk 1 above)
Comment (ahnitz) [inline bin/hdfcoinc/pycbc_make_coinc_search_workflow]: Ok, ignore this, I see you are placing everything under workflow, which also makes sense.

File: bin/hdfcoinc/pycbc_make_coinc_search_workflow (hunk 2 above)
Comment (duncan-brown) [inline bin/hdfcoinc/pycbc_make_coinc_search_workflow]: I'm changing this per @spxiwh's suggestion to put the files directly under the results directory when they are created.

**Comment by ahnitz** [comment]
Ok, a general comment. There are really 3 pull requests worth of changes here, as they are mostly separable changes. If possible please try to split these kind of changes into smaller units in the future.

This could have been broken up into 
* missing version from executables
* add protect option to make html page exe
* add additional logging to workflow and separate information into workflow results tab.  

For now, I think the changes are mostly OK, except for the few places I've comment. The one place I have general concerns is with how complicated the submit script is getting. I think we should seriously consider converting this to a python script at this point with some associated helper functions/modules. I'll leave that for future though/discussion.

**Comment by cmbiwer** [comment]
I guess I'll add that since you're writing segment XML files to the ``results/`` subdir there should be some basic HTML table rendering.

I didn't see the segment XML files in Duncan's original example (which looked like it should have been ``analysis_time/segment_data``) but whenever someone does a new run this would be something to check.

**Comment by duncan-brown** [comment]
@ahnitz yes, this grew into a much bigger pull request that I originally expected, so fair comment. I will try and keep it smaller in future.

I agree that ```pycbc_submit_dax``` is getting too complicated. We should do a complete re-write in Python for a future release, but that should also taking into account DIBBS development.

**Comment by duncan-brown** [comment]
Running a full check and then I will merge.

**Pull request merged**